GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -431,6 +434,7 @@ private void Update()
                         bool isSuccess = false;
                         try
                         {
+                            await TaskScheduler.Default;
```

Comment:
❔ Why is this changing? I would expect `GitCommandHelpers.GetAllChangedFilesCmd` to return effectively immediately, and `GetOutputAsync` to be properly asynchronous. If part of this assumption does not hold, can you clarify which part and why?

Reply:
I expected it to work as you described too, but the Git command is executing on the UI thread according to the GE Git Command Log
(is that presentation incorrect?)

Reply:
> but the Git command is executing on the UI thread

The git command _starts_ execution on the UI thread, but as long as the process is executed asynchronously the UI thread will not wait for the process to complete. Starting a process from the UI thread should not cause an observable delay. Waiting for a process to complete on the the UI thread is the problematic operation.

Reply:
Can the presentation be changed to include this, is it available for the command?
ThreadHelper.JoinableTaskContext.IsOnMainThread is always false after execution also if run synchronously.